.travis.yml

```diff
@@ -1,39 +1,83 @@
 language: cpp
+dist: trusty
 
-compiler:
-  - gcc
-  - clang
+matrix:
```

Comment:
Why such complex matrix, compared to the simple one we had before?
Also, before we did not have versions of gcc / clang fixed, which on one hand meant they could change I guess (or was it tied to distribution)?
I prefer simple configuration over complex one, if complex one does not bring much extra benefits, but I am sure you have reasons for this so I would love to hear them!

Reply:
I generally agree with simple configurations. The problem with toolchains is that Travis doesn't have a notion of "latest", i.e., there's no way to get GCC 9 now right now automagically. By specifying a large range of compilers, you want to cover as much real estate for `-Werror` coverage

Reply:
Ok, so it just uses whatever comes by default with the distribution, and you wanted to also test some newer versions of these compilers, that is it?
If that is so, ok, makes sense! I wish there was easier way then providing so much configuration, but I understand there is not. Could you maybe add a comment, explaining that we are doing this in order to be able to test specific versions of compilers? Or do you think it is redundant?

Reply:
Yes exactly. The reason why Travis does this is pure stability. There are tons and tons of C++ codebases that, upon upgrading to GCC 8 or 9, will instantly fall over, due to GCC becoming stricter with every release. Imagine your CI falling over all of a sudden, just because a new GCC was released and Travis decided to give you that now. People would be protesting against them with pitchforks.

I'll add a comment.

Reply:
@Martinsos added a comment now

Reply:
Makes perfect sense! All good.